llvm/lib/Target/PowerPC/PPCMergeStringPool.cpp

```diff
@@ -117,9 +118,16 @@ class PPCMergeStringPool : public ModulePass {
 // sure that they can be replaced.
 static bool hasReplaceableUsers(GlobalVariable &GV) {
   for (User *CurrentUser : GV.users()) {
-    // Instruction users are always valid.
-    if (isa<Instruction>(CurrentUser))
+    if (isa<Instruction>(CurrentUser)) {
+      if (auto *II = dyn_cast<IntrinsicInst>(CurrentUser)) {
+        // Some intrinsics require a plain global.
+        if (II->getIntrinsicID() == Intrinsic::eh_typeid_for)
+          return false;
+      }
+
```

Comment:
Hm, maybe adding this check is a better fix. I assume it will also cover the eh.typeid.for issue if this intrinsic is only used if the global is *also* mentioned in a landingpad.

Reply:
I went ahead and added the EHPad check while also keeping the explicit eh.typeid.for check. Better safe than sorry...

Reply:
Fair enough.